Update project to use latest stable and 2018

jacobgardner opened the pull request:
Since [1.30](https://blog.rust-lang.org/2018/10/25/Rust-1.30.0.html#procedural-macros) proc-macro has been in stable rust and also includes things such as proper span information on tokens.  I don't think there anything left in this crate that needs nightly any more which should allow more developers to use this.  We also don't need macro_use either.  

With [2018 edition](https://rust-lang-nursery.github.io/edition-guide/rust-2018/module-system/path-clarity.html) there's a bunch of additional path cleanup we can do in this crate.  I probably missed some of the fixes, but it cleans up a lot of the boilerplate at the top of the program.  

Let me know if there's any issues.

jacobgardner committed:
Update project to use latest stable and 2018

jacobgardner committed:
Missed some files and completely flumoxed some with another branch

jacobgardner committed:
And I accidentally commited the bad snapshots

anp commented on cargo-snapshot/src/main.rs:
```diff
@@ -23,6 +10,9 @@ use duct::cmd;
 use walkdir::WalkDir;
 
+use error_chain::*;
+use clap::*;
```
Can we avoid glob imports here?

anp commented on cargo-snapshot/src/parse_tests.rs:
```diff
@@ -1,7 +1,7 @@
 
-use nom;
 use nom::{line_ending, digit, space};
+use nom::*;
```
same re: globs

anp reviewed (changes requested):
Hi! Thanks so much for the PR :D.

I had a couple of minor nits about glob imports (I know I committed the original sin, but by sending PRs you've demonstrated this code may be valuable and therefore I was wrong to do so!), otherwise this is awesome.

jacobgardner committed:
Remove almost all the globs

jacobgardner commented:
I removed all the globs but the prelude one in there.  I reverted back to using #[macro_use] for some of the crates because the macros had other macro dependencies which was making the dependency list quite long.  

This issue will go away when the crates start using $crate::macro_name, but for now, I thought this made the most sense.

jacobgardner committed:
Fix broken tests

jacobgardner committed:
Quote has the same problem as the other crates do where we need all the macro dependencies

anp commented:
Thank you!
